src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs

```diff
@@ -3628,6 +3628,46 @@ public async Task TestDuplicateProjectReference()
             }
         }
 
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public async Task TestEditorConfigDiscovery()
+        {
+            var files = GetSimpleCSharpSolutionFiles()
+                .WithFile(@"CSharpProject\CSharpProject.csproj", Resources.ProjectFiles.CSharp.WithDiscoverEditorConfigFiles)
+                .WithFile(".editorconfig", "root = true");
+
+            CreateFiles(files);
+
+            using (var workspace = CreateMSBuildWorkspace())
+            {
+                var projectFullPath = GetSolutionFileName(@"CSharpProject\CSharpProject.csproj");
+
+                var project = await workspace.OpenProjectAsync(projectFullPath);
+                var analyzerConfigDocument = Assert.Single(project.AnalyzerConfigDocuments);
+                Assert.Equal(".editorconfig", analyzerConfigDocument.Name);
+                var text = await analyzerConfigDocument.GetTextAsync();
+                Assert.Equal("root = true", text.ToString());
+            }
+        }
+
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public async Task TestEditorConfigDiscoveryDisabled()
+        {
+            var files = GetSimpleCSharpSolutionFiles()
+                .WithFile(@"CSharpProject\CSharpProject.csproj", Resources.ProjectFiles.CSharp.WithDiscoverEditorConfigFiles)
+                .ReplaceFileElement(@"CSharpProject\CSharpProject.csproj", "DiscoverEditorConfigFiles", "false")
```

Comment:
@jasonmalinowski Are we planning to provide an API on [ProjectInfo](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/ProjectInfo.cs,b954394cb3960811) to query the value of `DiscoverEditorConfigFiles` for a given project? For example, I think I need to guard the code action being added in https://github.com/dotnet/roslyn/pull/35691 to only show up if `DiscoverEditorConfigFiles = true`.

Reply:
We weren't planning on it -- the flag is more a feature flag than anything else and we only expect people where it's a problem to turn it off once we roll the feature out.

Why would that code action only work if discovery is on? If discovery is off either (a) you have no .editorconfig files, or (b) you have discovery off but manually specified some in your project. In the (b) case the code fix should still work, right?

Reply:
> Why would that code action only work if discovery is on?

If the code action adds an entry `dotnet_diagnostic.CAxxxx.severity = severity`, will it be respected? I think for the code style based severity configuration, it will still be respected in live analysis.
Regardless, I presume I need to test this scenario more thoroughly to ensure sanity.

Reply:
It'd be respected if [a] there are analyzer config documents in the project (even if they weren't discovered nicely) and [b] you have the feature flag that @JoeRobich is updating.